src/primitives/triangle/mod.rs

```diff
@@ -185,33 +175,35 @@ impl Triangle {
     /// This method can be used to determine if the triangle is colinear by checking if the returned
     /// value is equal to zero.
     pub(in crate::primitives) fn area_doubled(&self) -> i32 {
-        let Self { p1, p2, p3 } = self;
+        let [p1, p2, p3] = self.vertices;
 
         -p2.y * p3.x + p1.y * (p3.x - p2.x) + p1.x * (p2.y - p3.y) + p2.x * p3.y
     }
 
     /// Create a new triangle with points sorted in a clockwise direction.
     pub(in crate::primitives::triangle) fn sorted_clockwise(&self) -> Self {
-        let Self { p1, p2, p3 } = self;
+        let [p1, p2, p3] = self.vertices;
 
         let determinant = -p2.y * p3.x + p1.y * (p3.x - p2.x) + p1.x * (p2.y - p3.y) + p2.x * p3.y;
 
         match determinant.cmp(&0) {
             // Triangle is wound CCW. Swap two points to make it CW.
-            Ordering::Less => Self::new(*p2, *p1, *p3),
+            Ordering::Less => Self::new(p2, p1, p3),
```

Comment:
After the other changes are applied this is the only place that still uses `p1`...`p3`, so I would suggest to simply change it to this:
```suggestion
            Ordering::Less => Self::new(self.vertices[1], self.vertices[0], self.vertices[2]),
```
Or move the `let [p1, p2, p3] = self.vertices;` into the `Less` match arm, if you prefer that.